### ISSUE TITLE
Crash with modindex_common_prefix

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#864]
I encounter the following crash when i set `modindex_common_prefix` to a prefix that doesn't include a trailing dot, like `['foo']`, instead of `['foo.']`.

```
# Sphinx version: 1.1.2
# Python version: 2.7.2+
# Docutils version: 0.8.1 release
# Jinja2 version: 2.6
Traceback (most recent call last):
  File "<...>/lib/python2.7/site-packages/sphinx/cmdline.py", line 189, in main
    app.build(force_all, filenames)
  File "<...>/lib/python2.7/site-packages/sphinx/application.py", line 204, in build
    self.builder.build_update()
  File "<...>/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 196, in build_update
    'out of date' &#37; len(to_build))
  File "<...>/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 252, in build
    self.write(docnames, list(updated_docnames), method)
  File "<...>/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 283, in write
    self.prepare_writing(docnames)
  File "<...>/lib/python2.7/site-packages/sphinx/builders/html.py", line 267, in prepare_writing
    content, collapse = indexcls(domain).generate()
  File "<...>/lib/python2.7/site-packages/sphinx/domains/python.py", line 528, in generate
    entries[-1][1] = 1
IndexError: list index out of range
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/864
- Originally reported by: Piet Delport
- Originally created at: 2012-01-26T20:07:22.533

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2012-01-29 09:43:37+00:00_

Closes #864: Fix crash with some settings of :confval:`modindex_common_prefix`.

→ <<cset a6512317bfb9946ba700daa5163b66e69de65b4c>>

**Comment by shimizukawa** [comment]
_From Piet Delport on 2012-01-29 20:18:46+00:00_

Thanks!
